bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,921 @@
+            ts_date: new Date(parseInt(TimeSheetDateTimeStamp)),
+            person_work_agreement: timeEntry.PersonWorkAgreement,
+            person_work_agreement_external_id: timeEntry.PersonWorkAgreementExternalID,
+            ts_df_controlling_area: timeEntry.TimeSheetDataFields.ControllingArea,
+            ts_df_hours_unit_of_measure: timeEntry.TimeSheetDataFields.HoursUnitOfMeasure,
+            ts_df_purchase_order_item: timeEntry.TimeSheetDataFields.PurchaseOrderItem,
+            ts_df_receiver_cost_center: timeEntry.TimeSheetDataFields.ReceiverCostCenter,
+            ts_df_recorded_hours: timeEntry.TimeSheetDataFields.RecordedHours,
+            ts_df_recorded_quantity: timeEntry.TimeSheetDataFields.RecordedQuantity,
+            ts_df_ts_overtime_category: timeEntry.TimeSheetDataFields.TimeSheetOvertimeCategory,
+            ts_df_ts_task_component: timeEntry.TimeSheetDataFields.TimeSheetTaskComponent,
+            ts_df_ts_task_level: timeEntry.TimeSheetDataFields.TimeSheetTaskLevel,
+            ts_df_ts_task_type: timeEntry.TimeSheetDataFields.TimeSheetTaskType,
+            ts_df_ts_note: timeEntry.TimeSheetDataFields.TimeSheetNote,
+            ts_operation: timeEntry.TimeSheetOperation,
+            ts_status: timeEntry.TimeSheetStatus,
+        })
+    }
+
+    dataStore.save("TimeEntries", timeEntries);
+    dataStore.save("TimeEntriesClock", timeEntries);
+}
+
+async function createTimeEntry({dataStore, client, actionParameters}) {
+
+    const headers = await createHeaderPost(client)
+    const isTimeClockManagement = false
+    const date = moment(actionParameters.Date).valueOf()
+    const recordedTime = timeToDecimal(actionParameters.RecordedQuantity)
+    const employeeBusinessPartner = await getEmployeeBusinessPartner(client, actionParameters.UserEmail)
+    const workAgreement = await getWorkAgreement(client, employeeBusinessPartner)
+
+    const createTimeEntryApiRequestObject = {
+        "headers": headers,
+        "client": client,
+        "receiverCostCenter": workAgreement.CostCenter,
+        "purchaseOrderItem": actionParameters.PurchaseOrderItem,
+        "timeSheetTaskType": actionParameters.TimeSheetTaskType,
+        "timeSheetTaskLevel": actionParameters.TimeSheetTaskLevel,
+        "timeSheetTaskComponent": actionParameters.TimeSheetTaskComponent,
+        "timeSheetNote": actionParameters.TimeSheetNote,
+        "recordedHours": recordedTime.toString(),
+        "hoursUnitOfMeasure": actionParameters.HoursUnitOfMeasure,
+        "personWorkAgreementExternalID": workAgreement.PersonWorkAgreementExternalID,
+        "companyCode": workAgreement.CompanyCode,
+        "personWorkAgreement": workAgreement.PersonWorkAgreement,
+        "date": date,
+        "timeSheetStatus": actionParameters.TimeSheetStatus,
+        "timeSheetOperation": actionParameters.TimeSheetOperation,
+        "timeEntryId": actionParameters.TimeEntryId,
+        "startedAt": null,
+        "endAt": '',
+    }
```

Comment:
Please, do the same here as with `storeObject`.
1. extract `headers` and `client` as separate arguments. They shouldn't be in `createTimeEntryApiRequestObject`
2. Create a new function `getTimeEntryApiRequestObject(workAgreement, actionParameters)`
3. Use this function in `createTimeEntry()` and in `clockManager()`

```js
const createTimeEntryApiRequestObject = {
       ...getTimeEntryApiRequestObject(workAgreement, actionParameters),
       date,
       startedAt: null,
       // etc. etc.
}

storeEntryInDatabase(client, headers, storeObject)
```

Reply:
new function created and used on createTimeEntry() and clockManager()